From 360ee399b72a8688f86b8a258add41f162748d15 Mon Sep 17 00:00:00 2001
From: techyminati <techyminati@outook.com>
Date: Mon, 14 Jul 2025 12:13:04 +0000
Subject: [PATCH] Settings: Hack Framework to show correct max value

* For some reason, MAX CAPACITY reading is broken with mtk fg, so this is a dirty hack
 to always use DESIGN_CAPACITY in all scenarios and hence return 100%

Change-Id: I0810588407a1855bdd746581ad20b08e3d8d26ae
---
 ...ryMaximumCapacityPreferenceController.java | 33 +++++++++----------
 1 file changed, 15 insertions(+), 18 deletions(-)

diff --git a/src/com/android/settings/deviceinfo/batteryinfo/BatteryMaximumCapacityPreferenceController.java b/src/com/android/settings/deviceinfo/batteryinfo/BatteryMaximumCapacityPreferenceController.java
index 186945efde8..501b9bc3b12 100644
--- a/src/com/android/settings/deviceinfo/batteryinfo/BatteryMaximumCapacityPreferenceController.java
+++ b/src/com/android/settings/deviceinfo/batteryinfo/BatteryMaximumCapacityPreferenceController.java
@@ -38,23 +38,20 @@ public class BatteryMaximumCapacityPreferenceController extends BasePreferenceCo
         return AVAILABLE;
     }
 
-    @Override
-    public CharSequence getSummary() {
-        Intent batteryIntent = BatteryUtils.getBatteryIntent(mContext);
-        final int maxCapacityUah =
-                batteryIntent.getIntExtra(BatteryManager.EXTRA_MAXIMUM_CAPACITY, -1);
-        final int designCapacityUah =
-                batteryIntent.getIntExtra(BatteryManager.EXTRA_DESIGN_CAPACITY, -1);
-
-        if (maxCapacityUah > 0 && designCapacityUah > 0) {
-            int maxCapacity = maxCapacityUah / 1_000;
-            int designCapacity = designCapacityUah / 1_000;
-            int percentage = (maxCapacity * 100) / designCapacity;
-
-            return mContext.getString(
-                    R.string.battery_maximum_capacity_summary, maxCapacity, percentage);
-        }
-
-        return mContext.getString(R.string.battery_maximum_capacity_not_available);
+@Override
+public CharSequence getSummary() {
+    Intent batteryIntent = BatteryUtils.getBatteryIntent(mContext);
+    final int designCapacityUah =
+            batteryIntent.getIntExtra(BatteryManager.EXTRA_DESIGN_CAPACITY, -1);
+
+    if (designCapacityUah != -1) {
+        int designCapacity = designCapacityUah / 1_000;
+        int percentage = 100;
+
+        return mContext.getString(
+                R.string.battery_maximum_capacity_summary, designCapacity, percentage);
     }
+
+    return mContext.getString(R.string.battery_maximum_capacity_not_available);
+}
 }
-- 
2.34.1

